src/clusterfuzz/_internal/cron/job_exporter.py

```diff
@@ -0,0 +1,160 @@
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Exports Jobs, Fuzzer, DataBundles and their JobTemplates to GCS, 
+    in order to mirror the workloads on testing environments."""
+
+import os
+
+from google.cloud import ndb
+from google.protobuf import any_pb2
+
+from clusterfuzz._internal.bot.tasks.utasks import uworker_io
+from clusterfuzz._internal.datastore import data_types
+from clusterfuzz._internal.google_cloud_utils import blobs
+from clusterfuzz._internal.google_cloud_utils import gsutil
+from clusterfuzz._internal.google_cloud_utils import storage
+from clusterfuzz._internal.metrics import logs
+
+target_entities = [
+    (data_types.Fuzzer, ['blobstore_key', 'sample_testcase'], 'fuzzer'),
+    (data_types.Job, ['custom_binary_key'], 'job'),
+    (data_types.DataBundle, [], 'databundle'),
+    (data_types.JobTemplate, [], 'jobtemplate'),
+]
+
+
+class RSyncClient:
+  """Interface that defines the rsync contract."""
+
+  def __init__(self):
+    pass
+
+  def rsync(self, source: str, target: str):
+    pass
+
+
+class GCloudCLIRSync(RSyncClient):
+  """RSyncClient implementation that delegates to the gcloud cli.
+    Unsuitable for unit testing."""
+
+  def __init__(self):
+    self._runner = gsutil.GSUtilRunner()
+
+  def rsync(self, source: str, target: str):
+    self._runner.rsync(f'gs://{source}', target)
+
+
+class StorageRSync(RSyncClient):
+  """RSyncClient implementation for unit testing,
+    meant for use with GCS emulator."""
+
+  def __init__(self):
+    pass
+
+  def rsync(self, source: str, target: str):
+    for blob in storage.list_blobs(f'gs://{source}'):
+      blob_target_path = f'{target}/{blob}'
+      storage.copy_blob(f'gs://{source}/{blob}', blob_target_path)
+
+
+class EntityMigrator:
+  """Serializes entities to GCS, and imports them back."""
+
+  def __init__(self, target_cls: ndb.Model, blobstore_keys: list[str],
+               entity_type: str, rsync_client: RSyncClient, export_bucket: str):
+    self._target_cls = target_cls
+    self.blobstore_keys = blobstore_keys
+    self._entity_type = entity_type
+    self._rsync_client = rsync_client
+    self._export_bucket = export_bucket
+
+  def _serialize(self, entity) -> bytes:
+    return uworker_io.entity_to_protobuf(entity).SerializeToString()
+
+  def _deserialize(self, proto_as_str: bytes) -> ndb.Model:
+    deserialized_any = any_pb2.Any()  # pylint: disable=no-member
+    # Parse the bytes into the Any message
+    deserialized_any.ParseFromString(proto_as_str)
+    return uworker_io.entity_from_protobuf(deserialized_any, self._target_cls)
```

Comment:
Sort of a weird dependency. I'm fine with it. But maybe it would be better to put this function in data_handler (also a bad module).